week2/project/ecommerce/src/components/CategoryList.jsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react'
+
+const CategoryList = ({ selectedCategory, setSelectedCategory, setProducts }) => {
```

Comment:
```suggestion
const CategoryList = ({ setProducts }) => {
  const [selectedCategory, setSelectedCategory] = useState("");
```

Refactored selectedCategory to be an internal state in CategoryList since it’s only used for UI state within this component. This simplifies the parent component and makes CategoryList more self-contained.